apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowKey.cs

```diff
@@ -56,6 +58,38 @@ public RowKey(ByteString key)
 
         // TODO: Docs
 
+        // TODO: Make sure making RowKey comparable is useful.
```

Comment:
One potential use: if the server always returns the rows in RowKey order, it will make it easier to write tests if you can take a bunch of rows and order them (locally) by row key, then expect them to be returned like that by the server.

Reply:
Not sure about any ordering guarantee. I'll ask internally.

Reply:
So it looks the cells will sorted by row key, then family (by some internal id though, so this needs to be re-sorted lexicographically by this library), then column qualifier, then timestamp/version descending. So this could be comparable for testing and so could `ColumnQualifier`. The more I think of it, the more I think it makes sense to merge these types into one and then use the new type for cell values as well. They are pretty much identical. `BigtableBytestring` perhaps? Also, I think it makes sense to make it a struct to reduce any additional GC pressure when dealing with a large numbers of cells. WDYT?

Reply:
+1 for using a struct to wrap the ByteString. (I'm creating a Blob struct in Firestore, but I'm not making that comparable yet as I don't need to.)

I don't understand the domain well enough to judge whether one type is better than three here or not... I'd go with your gut, and we can always change our minds. (Also, check what Java does.)

Reply:
Java seems to always uses byte[] or ByteString. The samples show all row keys being sent to `Bytes.toBytes` before making requests. Go seems to take string always. And python looks like it can take bytes or strings for the same parameter and internally converts if necessary.

I feel like it make sense to merge these. The whole point is so we can introduce implicit operators so you don't have to see the type name of `ByteString` or whatever these are called. The only possible thing which might be useful for one and not the others is a method/operator to convert the byte string to/from a long for cases where a cell value can be treated like a 8-byte, big-endian encoded signed integer. It would make getting/setting the current value in those cells much easier. But I don't think having those unnecessary members would hurt the row key or column qualifiers in any way.